tests/django_pg_migration_tools/test_operations.py

```diff
@@ -1692,8 +1693,102 @@ def test_when_not_allowed_to_migrate_by_the_router(self):
+
+        assert queries[0]["sql"] == dedent("""
+            SELECT 1
+            FROM pg_catalog.pg_attribute
+            WHERE
+                attrelid = 'example_app_nullfkfieldmodel'::regclass
+                AND attname = 'fk_id'
+                AND attnotnull IS TRUE;
+        """)
+        assert queries[1]["sql"] == dedent("""
+            SELECT conname
+            FROM pg_catalog.pg_constraint
+            WHERE conname = 'example_ap_fk_id_9fd70957e5';
+        """)
+        assert queries[2]["sql"] == dedent("""
+            ALTER TABLE "example_app_nullfkfieldmodel"
+            ADD CONSTRAINT "example_ap_fk_id_9fd70957e5"
+            CHECK ("fk_id" IS NOT NULL) NOT VALID;
+        """)
+        assert queries[3]["sql"] == dedent("""
+            ALTER TABLE "example_app_nullfkfieldmodel"
+            VALIDATE CONSTRAINT "example_ap_fk_id_9fd70957e5";
+        """)
+        assert queries[4]["sql"] == dedent("""
+            ALTER TABLE "example_app_nullfkfieldmodel"
+            ALTER COLUMN "fk_id"
+            SET NOT NULL;
+        """)
+        assert queries[5]["sql"] == dedent("""
+            ALTER TABLE "example_app_nullfkfieldmodel"
+            DROP CONSTRAINT "example_ap_fk_id_9fd70957e5";
+        """)
+
+        with connection.schema_editor(atomic=False, collect_sql=False) as editor:
+            with utils.CaptureQueriesContext(connection) as reverse_queries:
+                operation.database_backwards(
+                    self.app_label, editor, from_state=new_state, to_state=project_state
+                )
+        assert len(reverse_queries) == 2
+
+        assert reverse_queries[0]["sql"] == dedent("""
+            SELECT 1
+            FROM pg_catalog.pg_attribute
+            WHERE
+                attrelid = 'example_app_nullfkfieldmodel'::regclass
+                AND attname = 'fk_id'
+                AND attnotnull IS TRUE;
+        """)
+        assert reverse_queries[1]["sql"] == dedent("""
+            ALTER TABLE "example_app_nullfkfieldmodel"
+            ALTER COLUMN "fk_id"
+            DROP NOT NULL;
+        """)
+
+        # Reversing again does nothing apart from checking the field is already
+        # nullable.
+        with connection.schema_editor(atomic=False, collect_sql=False) as editor:
+            with utils.CaptureQueriesContext(connection) as second_reverse_queries:
+                operation.database_backwards(
+                    self.app_label, editor, from_state=new_state, to_state=project_state
+                )
+        assert len(second_reverse_queries) == 1
+
+        assert second_reverse_queries[0]["sql"] == dedent("""
+            SELECT 1
+            FROM pg_catalog.pg_attribute
+            WHERE
+                attrelid = 'example_app_nullfkfieldmodel'::regclass
+                AND attname = 'fk_id'
+                AND attnotnull IS TRUE;
+        """)
+
+    @pytest.mark.django_db(transaction=True)
+    def test_when_field_is_a_foreign_key(self):
```

Comment:
I think this test name got switched with the one for the new test above.

Reply:
It did! Will amend.